File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.length()==0) return 0;
        int maxiLength = INT_MIN;
        int size = s.length();
        int left=0;
        unordered_set<char>st;

        for(int right=0;right<size;right++){
            while(st.find(s[right])!=st.end()){
                st.erase(s[left]);
                left++;
            }
            st.insert(s[right]);
            maxiLength = max(maxiLength, right-left+1);
        }
        return maxiLength;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
 public int lengthOfLongestSubstring(String s) {
    int ans = 0;
    HashSet<Character> st = new HashSet<>();
    int left = 0;
    for(int right = 0; right < s.length(); right++){

        while(st.contains(s.charAt(right))){ // duplicate
            st.remove(s.charAt(left));
            left++;
        }

        st.add(s.charAt(right));
        ans = Math.max(ans, right - left + 1);
    }
    return ans;
    }
}